engine/src/main/java/org/hibernate/search/backend/impl/lucene/MultiWriteDrainableLinkedList.java

```diff
@@ -37,17 +37,39 @@
 	 */
 	public void add(T element) {
 		final Node<T> newnode = new Node<T>( element );
-		addNode( newnode );
+		addNode( newnode, newnode );
 	}
 
-	private synchronized void addNode(Node<T> newnode) {
+	/**
+	 * Adds multiple new entries to this list, in the given order,
+	 * making sure no entry is added between the given elements.
+	 *
+	 * @param elements The elements to add.
+	 */
+	public void addAll(Iterable<T> elements) {
```

Comment:
yes the `monitor` looks odd. I wouldn't remove it though, I suspect it's needed for Infinispan - and seems handy for us as well.

Feel free to create a new, similar class for Elasticsearch if you feel it doesn't fit as-is.